client/app/dbaas/logs/detail/options/logs-options.service.js

```diff
@@ -0,0 +1,154 @@
+        this.OvhApiDbaasLogs = OvhApiDbaasLogs;
+        this.LogOptionConstant = LogOptionConstant;
+    }
+
+    /**
+     * returns the transformed option. Meant to be used for the available options
+     *
+     * @param {any} option
+     * @returns the transformed option
+     * @memberof LogsOptionsService
+     */
+    _transformOption (option) {
+        option.quantity = 0;
+        option.price = option.prices[0].price.value;
+        option.priceText = option.prices[0].price.text;
+        option.type = this.$translate.instant(`${option.planCode}-type`);
+        option.detail = this.$translate.instant(`${option.planCode}-detail`);
+    }
+
+    /**
+     * returns the list of options available for selection
+     *
+     * @param {any} serviceName
+     * @returns promise which will be resolve to an array of options objects
+     * @memberof LogsOptionsService
+     */
+    getOptions (serviceName) {
+        return this.OvhApiOrderCartServiceOption.Lexi().get({
+            productName: this.LogOptionConstant.productName,
+            serviceName
+        }).$promise
+            .then(response => {
+                _.each(response, option => this._transformOption(option));
+                return response;
+            })
+            .catch(this.ServiceHelper.errorHandler("logs_options_options_loading_error"));
+    }
+
+    /**
+     * returns the transformed option, which has the count for each of the options.
+     * Meant to be used for the subscribed options
+     *
+     * @param {any} option
+     * @param {any} optionsCountMap
+     * @returns the transformed option
+     * @memberof LogsOptionsService
+     */
+    transformSubscribedOption (option, optionsCountMap) {
+        const optionConfig = {};
+        optionConfig.type = this.$translate.instant(`${option}-type`);
+        optionConfig.detail = this.$translate.instant(`${option}-detail`);
+        optionConfig.quantity = optionsCountMap[option];
+        return optionConfig;
+    }
+
+    /**
+     * makes API call to get the list of options that have been subscribed in the service
+     *
+     * @param {any} serviceName
+     * @returns promise that resolves with the array of options which have been subscribed
+     * @memberof LogsOptionsService
+     */
+    getSubscribedOptions (serviceName) {
+        return this.OvhApiDbaasLogs.Accounting().Aapi().me({
+            serviceName
+        })
+            .$promise
+            .catch(this.ServiceHelper.errorHandler("logs_options_current_options_loading_error"));
+    }
+
+    /**
+     * returns map of all subscribed options wiht there count.
+     *
+     * @param {any} serviceName
+     * @returns map of subscribed option with count
+     * @memberof LogsOptionsService
+     */
+    getSubscribedOptionsMap (serviceName) {
+        return this.getSubscribedOptions(serviceName)
+            .then(response => {
+                const optionsCountMap = _.reduce(response.options, (optionsMap, option) => {
```

Comment:
I can't say I'm sure what this code is doing after several reads.  Try to make it more clear if you can.  If not, a comment might be nice

Reply:
This change got into the PR because of merge conflicts. But I have commented the code to make it clearer